Index: bcfg2/src/lib/Bcfg2/Server/Lint/Metadata.py
===================================================================
--- bcfg2.orig/src/lib/Bcfg2/Server/Lint/Metadata.py
+++ bcfg2/src/lib/Bcfg2/Server/Lint/Metadata.py
@@ -124,7 +124,7 @@ class Metadata(ServerPlugin):
             grpname = grp.get("name")
             if grpname in duplicates:
                 duplicates[grpname].append(grp)
-            elif len(grp.attrib) > 1:  # group has options
+            elif len(grp.attrib) > 1 and 'negate' not in grp.attrib:  # group has options
                 if grpname in groups:
                     duplicates[grpname] = [grp, groups[grpname]]
                 else:
