Index: bcfg2/src/lib/Bcfg2/Server/Lint/MergeFiles.py
===================================================================
--- bcfg2.orig/src/lib/Bcfg2/Server/Lint/MergeFiles.py
+++ bcfg2/src/lib/Bcfg2/Server/Lint/MergeFiles.py
@@ -118,10 +118,9 @@ class MergeFiles(Bcfg2.Server.Lint.Serve
         for cname, cdata in others:
             seqmatch = SequenceMatcher(None, fdata.data, cdata.data)
             # perform progressively more expensive comparisons
-            if seqmatch.real_quick_ratio() == 1.0:
+            if seqmatch.quick_ratio() == 1.0:
                 identical.add(cname)
             elif (
-                seqmatch.real_quick_ratio() > Bcfg2.Options.setup.threshold and
                 seqmatch.quick_ratio() > Bcfg2.Options.setup.threshold and
                 seqmatch.ratio() > Bcfg2.Options.setup.threshold):
                 similar.add(cname)
